### PR TITLE
Suppress exception if download fails and file is present

### PR DESCRIPTION
**Pull request by BjoernAtBosch**
This commit suppresses an exception is throw if `obtain_local_file_path` is called with an url, no internet connections is available (i.e. being *offline*), and a (cached) local file already exists.

### CODE REVIEW

File: velocitas_lib/__init__.py [hunk 1]
```diff
@@ -181,14 +189,14 @@ def obtain_local_file_path(
...
-    if os.path.isfile(download_path):
-        path, file = os.path.split(download_path)
-        parts = file.split(".", 1)
-        filename = f"{parts[0]}_1.{parts[1]}" if len(parts) > 1 else f"{parts[0]}_1"
 
-        download_path = os.path.join(path, filename)
+    try:
+        download_file(path_or_uri, download_path)
+    except requests.ConnectionError:
+        if have_internet_connection() or not os.path.exists(download_path):
```
Comment (MP91) [inline velocitas_lib/__init__.py:196]: if the file can't be downloaded to whatever reason but we have internet connection the exception is also raised if the file is available.

Should we maybe split this check to only through if the file is not available? 
```suggestion
        if have_internet_connection() and not os.path.exists(download_path):
```

File: velocitas_lib/__init__.py [hunk 2]
```diff
@@ -157,6 +157,14 @@ def is_uri(path: str) -> bool:
     return re.match(r"(\w+)\:\/\/(\w+)", path) is not None
 
 
+def have_internet_connection() -> bool:
+    try:
+        requests.head("http://www.google.com/", timeout=5)
```
Comment (MP91) [inline velocitas_lib/__init__.py:162]: don't know if this is the right url to check. even with regards to china. Maybe we can find some better one. And maybe we directly check an ip like 1.1.1.1 to also avoid dns issues?

File: velocitas_lib/__init__.py (hunk 1 above)
Comment (BjoernAtBosch) [reply velocitas_lib/__init__.py:196]: This is by intention: If there is an online connection, we should always try to download the file and give an error if the download fails - regardless if the file is present

File: velocitas_lib/__init__.py (hunk 2 above)
Comment (BjoernAtBosch) [reply velocitas_lib/__init__.py:162]: Yes, well. I'm also not so happy with this. But I think the "1.1.1.1 check" could cause issues if being behind a corporate proxy: I checked it within Bosch Corporate Network - it is not working. This is the reason, why I tried something on http basis.

**Review by MP91 — APPROVED** [review]
lgtm

**Pull request merged**